cli/operator/node.go

```diff
@@ -219,26 +219,21 @@ var StartNodeCmd = &cobra.Command{
 		validatorStore := nodeStorage.ValidatorStore()
 		// validatorStore = newValidatorStore(...) // TODO
 
-		var messageValidator validation.MessageValidator
-
-		if networkConfig.AlanFork() {
-			messageValidator = validation.New(
-				networkConfig,
-				validatorStore,
-				dutyStore,
-				signatureVerifier,
-				validation.WithLogger(logger),
-				validation.WithMetrics(metricsReporter),
-			)
-		} else {
-			messageValidator = genesisvalidation.New(
+		messageValidator := validation.New(
+			networkConfig,
+			validatorStore,
+			dutyStore,
+			signatureVerifier,
+			validation.WithLogger(logger),
+			validation.WithMetrics(metricsReporter),
+			validation.WithGenesisValidator(genesisvalidation.New(
```

Comment:
As a small optimization, we can check if the fork happened and don't even create the genesis message validator in this case

Reply:
Agree. Maybe it even worths to do it inside the `WithGenesisValidator` option.